tests/unittests/Foundation/NSFileHandleTests.mm

```diff
@@ -16,330 +16,276 @@
 
 #import <TestFramework.h>
 #import <Foundation/Foundation.h>
-
+#import <Starboard/SmartTypes.h>
 #import <windows.h>
 #import <algorithm>
 
-NSString* getPathToFile(char const* filePath) {
-    size_t i = 0;
+static NSString* getModulePath() {
     char fullPath[_MAX_PATH];
-    int len = GetModuleFileNameA(NULL, fullPath, _MAX_PATH);
-
-// Path separator characters differ between platforms
-#if TARGET_OS_WIN32
-    const char slashChar = '\\';
-#else
-    const char slashChar = '/';
-#endif
-
-    char* ptrToLastEntry = strrchr(fullPath, slashChar);
-    int lengthOfLastEntry = strlen(ptrToLastEntry);
-
-    size_t destSize = _MAX_PATH - (len - lengthOfLastEntry);
-    size_t srcSize = strlen(filePath) + 1;
-
-// strncpy_s is a MSVC extension
-#if TARGET_OS_WIN32
-    strncpy_s(ptrToLastEntry, destSize, filePath, srcSize);
-#else
-    strncpy(ptrToLastEntry, filePath, std::min(destSize, srcSize));
-#endif
-
-    return [NSString stringWithFormat:@"%s", fullPath];
+    GetModuleFileNameA(NULL, fullPath, _MAX_PATH);
+    return [@(fullPath) stringByDeletingLastPathComponent];
 }
 
-void createFileWithContentAndVerify(char const* filePath, NSString* content) {
-    NSString* fullPath = getPathToFile(filePath);
-    NSFileHandle* fh = [NSFileHandle fileHandleForWritingAtPath:fullPath];
-    ASSERT_TRUE(fh != nil);
-
-    if (content) {
-        NSData* data = [content dataUsingEncoding:NSUTF8StringEncoding];
-        [fh writeData:data];
-    }
-
-    [fh closeFile];
+static NSString* getPathToFile(NSString* fileName) {
+    static StrongId<NSString*> refPath = getModulePath();
+    return [refPath stringByAppendingPathComponent:fileName];
+}
 
-    // verify file exists.
+static void createFileWithContentAndVerify(NSString* fileName, NSString* content) {
+    NSString* fullPath = getPathToFile(fileName);
+    NSError* error = nil;
+    ASSERT_TRUE([content writeToFile:fullPath atomically:NO encoding:NSUTF8StringEncoding error:&error]);
+    ASSERT_EQ(nil, error);
     ASSERT_TRUE([[NSFileManager defaultManager] fileExistsAtPath:fullPath]);
 }
 
-void deleteFile(char const* filePath) {
-    NSError* error;
-    ASSERT_TRUE_MSG([[NSFileManager defaultManager] removeItemAtPath:getPathToFile(filePath) error:&error],
-                    "FAILED:Unable to delete the test file.");
-    ASSERT_OBJCEQ(error, nil);
-}
+static void deleteFile(NSString* name) {
+    NSString* fullPath = getPathToFile(name);
+    if ([[NSFileManager defaultManager] fileExistsAtPath:fullPath]) {
+        [[NSFileManager defaultManager] removeItemAtPath:fullPath error:nil];
+    }
+};
 
-TEST(NSFileHandle, Init) {
-    ASSERT_TRUE_MSG([[[NSFileHandle alloc] init] autorelease] != nil, "FAILED: alloc/init failed.");
-}
+static void closeFile(NSFileHandle* handle) {
+    [handle closeFile];
+};
+
+using unique_fileDeleter = std::unique_ptr<NSString, decltype(&deleteFile)>;
+using unique_fileHandleCloser = std::unique_ptr<NSFileHandle, decltype(&closeFile)>;
 
```

Comment:
Make that _very_ instead of just _mildly_.

---
In reply to: [73436574](https://github.com/Microsoft/WinObjC/pull/805#discussion_r73436574) [](ancestors = 73436574)